### ISSUE TITLE
Staying Anonymous When Using Rdesktop

### ISSUE THREAD

**Issue by JohnMcKinsey** [rdesktop/rdesktop#259]
Hello,

my question is:

Is it possible to stay completely anonymous when using Rdesktop?

I am on Linux, I am thinking if I switch on a VPN with some lock feature that does not allow any traffic outside of the VPN tunnel, or if I manage to torify the traffic send between me and the remote computer via Rdesktop, I should be fine.

But then I heard of all the redirection features. So I want to ask if the voice, camera,  and all kinds of redirection are switched off by default right after you install Rdesktop, or whether some of it is turned on by default.

In other words, if I just install Rdesktop, connect it to the remote computer via the app through a VPN or Tor, and an adversary gets complete control over the remote computer, can he somehow find out who I am in reality (deanonymize me), or should I not be worried?

Or a better way to put, is the incoming IP address the only thing that the adversary would be able to read (from logs or when the session is active), or would there be other ways to find out who I am? As for the IP, if everything is setup correctly, the adversary should only be able to see the IP of the Tor exit guard or of the VPN server, as far as I understand it.

Thanks!
John

**Comment by uglym8** [comment]
Hi,

Regarding redirection: yes, it's off by default.
To redirect something you have to explicitly request such a redirection.

Reasoning about anonimity is a little bit trickier:)
I'd say that it's pretty safe to use it.
But to be on the safe side I'll check Microsoft RDP specs again.

Guys (@derfian, @hean01-cendio, @CendioOssman ) may be you could add some thoughts here?

**Comment by derfian** [comment]
rdesktop will transmit some information like cached device licensing, screen resolution, etc. to the server. A sufficiently motivated and capable adversary could probably use these "fingerprints" to help trace who used an account... maybe. On the other hand, if your identity is that interesting to someone, just get off the internet already.

Let me rephrase in a nicer way. If you have extraordinary strict requirements for security and anonymity, your best way forward would be to dig into both the rdesktop code and the RDP protocol specification to make your own judgements to see if they are acceptable. I don't see that ensuring perfect anonymity is the main focus of rdesktop.

**Comment by JohnMcKinsey** [comment]
Also, @derfian , could you please explain what you mean by "device licensing"? Like what kind of licensing would that be?

I read online that, indeed, screen size as well as the version of the OS and other pieces of information are sent to the server, but I am not sure what kind of licensing that would mean. Like license of the installed OS? Or the device ID, such as a laptop model number number? Or some license pertaining to Rdesktop itself? I would really welcome some elaboration.

Also, what do you guys think, could VNC be safer and more anonymous than RDP? For example, many VPS providers provide the possibility of connecting to the rented VPS either via RDP or VNC. Since VNC can be opened within a web browser (and that can be the anonymous Tor browser), I think it is generally much safer (since an app opened within the Tor browser would not have access to this information like screen size, installed OS, etc., or it would gather this info from the browser's user agent or from the size of browser window, both of which are kind of anonymized and obfuscated if you use the Tor browser, so the provided info would be cut to a minimum).

I am looking forward to hearing what you guys think.

**Comment by JohnMcKinsey** [comment]
Topic closed.
